experimental/FTheoryTools/src/auxiliary.jl

```diff
@@ -176,24 +176,59 @@ function _kodaira_type(id::MPolyIdeal{T}, f::T, g::T, d::T, ords::Tuple{Int64, I
     poly_g = ring_map(g)
     poly_d = ring_map(d)
     locus = ring_map(gens(id)[1])
+    locus_id = ideal([locus])
     
     if f_ord == 0 && g_ord == 0
-      monodromy_poly = _psi^2 + divexact(evaluate(9 * poly_g, [locus], [0]), evaluate(2 * poly_f, [locus], [0]))
+      g_quotient = divrem(9 * poly_g, locus)[2]
+      f_quotient = divrem(2 * poly_f, locus)[2]
+      quotient_val = div(g_quotient, f_quotient)
+
+      # quotient_gens = gens(groebner_basis(quotient(ideal([9 * poly_g]), ideal([2 * poly_f])) + locus_id))
+      # deleteat!(quotient_gens, findall(g -> g == locus, quotient_gens))
+      # quotient_val = quotient_gens[1]
+
+      monodromy_poly = _psi^2 + quotient_val
       kod_type = _string_from_factor_count(monodromy_poly, ["Non-split I_$d_ord", "Split I_$d_ord"])
     elseif d_ord == 4 && g_ord == 2 && f_ord >= 2
-      monodromy_poly = _psi^2 - evaluate(divexact(poly_g, locus^2), [locus], [0])
+      g_quotient = divrem(div(poly_g, locus^2), locus)[2]
+
+      # quotient_gens = gens(groebner_basis(ideal([div(poly_g, locus^2)]) + locus_id))
+      # deleteat!(quotient_gens, findall(g -> g == locus, quotient_gens))
+      # quotient_val = quotient_gens[1]
+
+      monodromy_poly = _psi^2 - g_quotient
       kod_type = _string_from_factor_count(monodromy_poly, ["Non-split IV", "Split IV"])
     elseif d_ord == 6 && f_ord >= 2 && g_ord >= 3
-      monodromy_poly =  _psi^3 + _psi * evaluate(divexact(poly_f, locus^2), [locus], [0]) + evaluate(divexact(poly_g, locus^3), [locus], [0])
+      f_quotient = divrem(div(poly_f, locus^2), locus)[2]
+      g_quotient = divrem(div(poly_g, locus^3), locus)[2]
+
+      # quotient_gens = gens(groebner_basis(ideal([_psi^3 + _psi * div(poly_f, locus^2) + div(poly_g, locus^3)]) + locus_id))
+      # deleteat!(quotient_gens, findall(g -> g == locus, quotient_gens))
+      
+      monodromy_poly = _psi^3 + _psi * f_quotient + g_quotient
       kod_type = _string_from_factor_count(monodromy_poly, ["Non-split I^*_0", "Semi-split I^*_0", "Split I^*_0"])
-    elseif f_ord == 2 && g_ord == 3 && d_ord >= 7 && d_ord % 2 == 1
-      monodromy_poly = _psi^2 + divexact(evaluate(divexact(poly_d, locus^d_ord) * divexact(2 * poly_f, locus^2)^3, [locus], [0]), 4 * evaluate(divexact(9 * poly_g, locus^3), [locus], [0])^3)
```

Comment:
With an eye towards migrating eventually into source and then also satisfying all `OSCAR` code standards - this line is pretty hard to read because it is so long. And having several combined calls to `divexact` and `evaluate` does not exactly make the reading any easier.

Suggestions:
Since the computation for the monodromy polynomial happens multiple times, maybe we could add an auxiliary function that completes this task? And this auxiliary function has strictly more than one line of code, maybe also a few comments on what exactly it does?

Reply:
The line you're commenting on here is removed in this PR, and I think the current version of the function doesn't have any excessively long lines in it. The reason I didn't add a helper function is because the monodromy polynomial is different for every singularity type. The bulk of the `_kodaira_type` function is this case-by-case computation of the monodromy polynomial anyway.

I do, however, agree that more comments would be helpful, I will add some in!

Reply:
Pardon. My bad.